src/utils/arrow.utils.ts

```diff
@@ -0,0 +1,41 @@
+import { useL10n } from '../hooks/useLocalization';
+import { ArrowItemType } from '../types/ArrowItemType';
+import { ArrowType } from '../types/ArrowType';
+import { TopicMapItemType } from '../types/TopicMapItemType';
+
+export const findItem = (
+  id: string,
+  items: Array<TopicMapItemType>,
+): TopicMapItemType | null => {
+  if (!id) {
+    return null;
+  }
+
+  return items.find((item) => item.id === id) ?? null;
+};
+
+export const getDescriptiveText = (
+  arrowItem: ArrowItemType,
+  items: Array<TopicMapItemType>,
+): string => {
+  const { startElementId, endElementId, arrowType } = arrowItem;
+
+  const startItem = findItem(startElementId, items);
+  const endItem = findItem(endElementId, items);
+
+  if (!startItem) {
+    throw new Error('Start item not found');
+  }
+  if (!endItem) {
+    throw new Error('End item not found');
+  }
+
+  const directionalLabel = useL10n('directionalArrowDescriptiveText');
+  const biDirectionalLabel = useL10n('biDirectionalArrowDescriptiveText');
```

Comment:
Beware of using hooks after if statements that throw or return early.

```suggestion
  const directionalLabel = useL10n('directionalArrowDescriptiveText');
  const biDirectionalLabel = useL10n('biDirectionalArrowDescriptiveText');

  if (!startItem) {
    throw new Error('Start item not found');
  }
  if (!endItem) {
    throw new Error('End item not found');
  }
```

Reply:
Also, we should pass the translator function to this function, to avoid using hooks here at all.

Reply:
(Or use `H5P.t()`)

Reply:
We can use your PR to add `H5P.t()`